packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1014,9 +1015,15 @@ export class Configuration {
     const configuration = new Configuration(startingCwd);
     configuration.importSettings(pickCoreFields(coreDefinitions));
 
+    const envList = rcFiles.map(rcFile => {
+      return [rcFile.cwd, configuration.getPackageEnv(rcFile.cwd)] as const;
+    });
+    const envMap = new Map<PortablePath, DotenvParseOutput>(envList);
+    configuration.values.set(`environment`, envMap);
+
     configuration.useWithSource(`<environment>`, pickCoreFields(environmentSettings), startingCwd, {strict: false});
     for (const {path, cwd, data} of rcFiles)
-      configuration.useWithSource(path, pickCoreFields(data), cwd, {strict: false});
+      configuration.useWithSource(path, pickCoreFields(data), cwd, {strict: false}, envMap.get(cwd));
```

Comment:
Shouldn't we have a single `env`, rather than use the one adjacent to each `.yarnrc.yml` file? The current design prevents using in `~/project/.yarnrc.yml` the variables declared in `~/.env`, for example

Reply:
You're right, i should find a way to make yarnrc.yml in a subdirectory able to read root's env.

I'm not quite sure in what context people would need two `yarnrc.yml` in one project, i guess if there are two `yarnrc.yml`, they might also need different `.env`, for example, a monorepo with a frontend project and a backend project that need different `.env`

Do you think I need to read the `.env` in the same folder in addition to reading the `.env` in the root directory? Or should I always just read the root project's `.env`?